utils/python/CIME/SystemTests/erp.py

```diff
@@ -11,134 +11,82 @@
-
-            # update the case to use the new values
-            self._case.read_xml()
-
-            # Use the second executable that was created
-            exeroot = self._case.get_value("EXEROOT")
-            cime_model = CIME.utils.get_model()
-            exefile  = os.path.join(exeroot,"%s.exe"%(cime_model))
-            exefile2 = os.path.join(exeroot,"%s.ERP%d.exe"%(cime_model,run))
-            if (os.path.isfile(exefile)):
-                os.remove(exefile)
-            shutil.copy(exefile2, exefile)
-
-            case_setup(self._case, test_mode=True, reset=True)
-            stop_n      = self._case.get_value("STOP_N")
-            stop_option = self._case.get_value("STOP_OPTION")
-
-            if run == 1:
-                expect(stop_n > 2, "ERROR: stop_n value %d too short"%stop_n)
-                rest_n = stop_n/2 + 1
-                self._case.set_value("REST_N", rest_n)
-                self._case.set_value("REST_OPTION", stop_option)
-                self._case.set_value("HIST_N", stop_n)
-                self._case.set_value("HIST_OPTION", stop_option)
-                self._case.set_value("CONTINUE_RUN", False)
-                suffix = "base"
-            else:
-                rest_n = stop_n/2 + 1
-                stop_new = stop_n - rest_n
-                expect(stop_new > 0, "ERROR: stop_n value %d too short %d %d"%(stop_new,stop_n,rest_n))
-                self._case.set_value("STOP_N", stop_new)
-                self._case.set_value("CONTINUE_RUN", True)
-                self._case.set_value("REST_OPTION","never")
-                suffix = "rest"
-            self.run_indv(suffix=suffix)
+        SystemTestsCompareTwo.__init__(self, case,
+                                       separate_builds = True,
+                                       run_two_suffix = "BFBRestTest",
+                                       run_one_st_archive = True,
+                                       run_one_description = "Baseline run from day 0 to STOP_N",
+                                       run_two_description = ("Test run starting from run one's " +
+                                                              "state at STOP_N // 2 + 1 to STOP_N.\n" +
+                                                              "Run 2 is done with half of the " +
+                                                              "number of tasks and threads as run 1"))
+
+    def _case_one_setup(self):
+        stop_n = self._case.get_value("STOP_N")
+        rest_n = _compute_rest_n(stop_n)
+        expect(stop_n > rest_n , "STOP_N value too small for test")
+        self._case.set_value("REST_N", rest_n)
+        self._case.set_value("REST_OPTION", self._case.get_value("STOP_OPTION"))
+
+    def _case_two_setup(self):
+        """ Case two uses half the number of threads and tasks as the defaults and case one. """
+        self._case.set_value("REST_OPTION", "never")
+
+        stop_n_1 = self._case1.get_value("STOP_N")
```

Comment:
It shouldn't hurt to get this from _case1, but it isn't necessary: you can get the old STOP_N value from the current case. I've been trying to write the other tests that way - i.e., only referring to the current self._case unless that's truly impossible. The SystemTestsCompareTwo infrastructure makes that possible by ensuring that case setup only happens once for each case - so (for example) you don't run the risk of re-halving the STOP_N value.